logstreams/src/main/java/io/zeebe/distributedlog/restore/impl/RestoreController.java

```diff
@@ -100,18 +99,29 @@ private void tryFindRestoreServer(CompletableFuture<MemberId> result) {
                 logReplicator,
                 snapshotReplicator,
                 response.getSnapshotRestoreInfo(),
-                latestLocalPosition,
-                backupPosition,
+                request.getLatestLocalPosition(),
+                request.getBackupPosition(),
                 server,
                 logger);
         result.complete(snapshotRestoreStrategy);
         break;
       case EVENTS:
-        result.complete(() -> logReplicator.replicate(server, latestLocalPosition, backupPosition));
+        logger.debug(
+            "Restoring events {} - {} from server {}",
+            request.getLatestLocalPosition(),
+            request.getBackupPosition(),
+            server);
+        result.complete(
+            () ->
+                logReplicator.replicate(
+                    server, request.getLatestLocalPosition(), request.getBackupPosition()));
         break;
       case NONE:
-        result.completeExceptionally(new UnsupportedOperationException("Not yet implemented"));
-        break;
+        logger.debug(
+            "Restore server {} reports nothing to replicate for request {}, finding new server...",
+            server,
+            request);
+        return buildRestoreStrategy(request);
```

Comment:
How about it returning a RestoreStrategy that doesn't do anything. We already have a retry logic in the primitive.
`result.complete(() -> latestLocalPosition)`

Reply:
That's a much better solution, actually :+1: